pubsub/envelope.go

```diff
@@ -84,14 +87,11 @@ func NewCloudEventsEnvelope(id, source, eventType, subject string, topic string,
 		PubsubField:          pubsubName,
 		TraceIDField:         traceID,
 		TraceStateField:      traceState,
+		AttributesField:      attributes,
 	}
 
 	ce[ceDataField] = ceData
 
-	if subject != "" {
```

Comment:
I think this is a breaking change.
Right now if we are not following the complete CE spec, I would much rather keep this field as well as add one in the attributes field if possible. Announce deprecation of the outer subject field and in 2 releases remove it. 
/cc @yaron2 @artursouza @pkedy

Reply:
Agree to keep this field temporarily, wait for more people to express their opinions, I am making changes.

Reply:
It does make sense to keep the current `NewCloudEventsEnvelope` method and store the optional, [`subject` attribute](https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#subject) in the attributes internally.

One question I have is why `attributes` is a `[]byte` instead of a `map[string]interface{}` to support the possible values from the CE spec?

Reply:
@pkedy 
The reason for using byte is that in the publish event, the definition of PublishEventRequest needs to be expressed by byte. The advantage of using []byte instead of map[string]interface{} is that dapr will not need to re-parse the attributes.
In dapr/dapr#3991 I gave the second step.